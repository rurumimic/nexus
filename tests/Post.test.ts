import { createTestContext } from './__helpers'

const ctx = createTestContext()

it('ensures that a draft can be created and published', async () => {
  // Create a new draft
  // Run operations against our API.
  const draftResult = await ctx.client.request(`
    mutation {
      createDraft(title: "Nexus", body: "...") {
        id
        title
        body
        published
      }
    }
  `)

  // Snapshot that draft and expect `published` to be false
  // The result will be snapshoted inline allowing us
  // to see the input and output collocated!
  expect(draftResult).toMatchSnapshot()

  // Publish the previously created draft
  const publishResult = await ctx.client.request(
    `
    mutation publishDraft($draftId: Int!) {
      publish(draftId: $draftId) {
        id
        title
        body
        published
      }
    }
    `,
    { draftId: draftResult.createDraft.id }
  )

  // Snapshot the published draft and expect `published` to be true
  expect(publishResult).toMatchSnapshot()

  const persistedData = await ctx.db.post.findMany()
  expect(persistedData).toMatchSnapshot()
})
