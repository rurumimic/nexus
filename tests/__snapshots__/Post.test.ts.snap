// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ensures that a draft can be created and published 1`] = `
Object {
  "createDraft": Object {
    "body": "...",
    "id": 1,
    "published": false,
    "title": "Nexus",
  },
}
`;

exports[`ensures that a draft can be created and published 2`] = `
Object {
  "publish": Object {
    "body": "...",
    "id": 1,
    "published": true,
    "title": "Nexus",
  },
}
`;

exports[`ensures that a draft can be created and published 3`] = `
Array [
  Object {
    "body": "...",
    "id": 1,
    "published": true,
    "title": "Nexus",
  },
]
`;
